donut/modules/bodfeedback/helpers.py

```diff
@@ -0,0 +1,248 @@
+            cursor.execute(query, complaint_id)
+            res = cursor.fetchall()
+        for row in res:
+            send_update_email(row['email'], complaint_id)
+
+
+def get_link(complaint_id):
+    '''
+    Gets a (fully qualified) link to the view page for this complaint id
+    '''
+    query = 'SELECT uuid FROM bod_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+        if not res:
+            return None
+        uuid = res['uuid']
+    return flask.url_for(
+        'bodfeedback.bodfeedback_view_complaint', id=uuid, _external=True)
+
+
+def get_id(uuid):
+    '''
+    Returns the complaint_id associated with a uuid
+    or false if the uuid is not found
+    '''
+    query = 'SELECT complaint_id FROM bod_complaint_info WHERE uuid = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, str(uuid))
+        if not cursor.rowcount:
+            return False
+    return cursor.fetchone()['complaint_id']
+
+
+def get_messages(complaint_id):
+    '''
+    Returns timestamps, posters, messages, and message_id's on this complaint
+    in ascending order of timestamp
+    '''
+    query = """
+    SELECT time, poster, message, message_id FROM bod_complaint_messages 
+    WHERE complaint_id = %s ORDER BY time
+    """
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchall()
+    if not (res and 'message_id' in res[0]):
+        return None
+    return res
+
+
+def get_summary(complaint_id):
+    '''
+    Returns a dict with the following fields: subject, status
+    '''
+    query = 'SELECT subject, status FROM bod_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        return cursor.fetchone()
+
+
+def get_subject(complaint_id):
+    '''
+    Returns the suject or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    return res['subject'] if res else None
+
+
+def get_status(complaint_id):
+    '''
+    Returns the status of a post or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    return res['status'] if res else None
+
+
+def mark_read(complaint_id):
+    '''
+    Sets the status of this complaint to 'read'
+    returns False if complaint_id is invalid
+    '''
+    query = "UPDATE bod_complaint_info SET status = 'read' WHERE complaint_id = %s"
+    if get_status(complaint_id) is None:
+        return False
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+
+
+def mark_unread(complaint_id):
+    '''
+    Sets the status of this complaint to 'new_msg'
+    returns False if complaint_id is invalid
+    '''
+    query = "UPDATE bod_complaint_info SET status = 'new_msg' WHERE complaint_id = %s"
```

Comment:
Should these statuses (stati?), e.g. `new_msg` and `read` be defined as enums somewhere?